### PR TITLE
Cleanup form groups and tabs properly

### PR DESCRIPTION
**Pull request by EmmanuelVella**
Currently the ``Admin::$formTabs`` variable is not cleaned up after calling ``$formMapper->remove()``. In this case, empty groups may be visible :

**Before**
before

**After**
after

To fix this, instead of adding a new method called after each ``remove()`` call, I moved all the logic into a single one, called when the formBuilder build is over.

### CODE REVIEW

File: Admin/Admin.php [hunk 1]
```diff
@@ -1620,20 +1653,6 @@ public function setFormGroups(array $formGroups)
     }
 
     /**
-     * {@inheritdoc}
-     */
-    public function removeFieldFromFormGroup($key)
```
Comment (rande) [inline Admin/Admin.php]: Why do you remove this method ?

Comment (EmmanuelVella) [inline Admin/Admin.php]: Because it was only used in the ``FormMapper::remove()`` method (see https://github.com/sonata-project/SonataAdminBundle/pull/2814/files#diff-90fea2ed781a41c220e9674b87a94c81L170) and is not called anymore, it is replaced with the new ``Admin::cleanupFormGroupsAndTabs`` method (see https://github.com/sonata-project/SonataAdminBundle/pull/2814/files#diff-81cd2da3c7ffebe278779869361e685aR1339)

Comment (rande) [inline Admin/Admin.php]: Methods can be called from an extension, so this is a big BC break. We need to keep them

Comment (EmmanuelVella) [inline Admin/Admin.php]: Yes in theory it can be called from an extension as its public, but I don't see what would be the use case ?

If you really want to keep it, maybe I can keep an empty method in the ``Admin`` class only ?

Comment (Soullivaneuh) [inline Admin/Admin.php]: > Methods can be called from an extension, so this is a big BC break. We need to keep them

Agree with @rande. As far as the method is public, it could be call anywhere and cause BC break if it is removed.

If this method is really not needed, please keep it with a deprecation mark (`@deprecated` + `trigger_error`) and make another PR to removed it on next major release.

Comment (EmmanuelVella) [inline Admin/Admin.php]: Thanks for the feedback. I have updated my PR accordingly.

File: Admin/Admin.php [hunk 2]
```diff
@@ -1625,6 +1658,8 @@ public function setFormGroups(array $formGroups)
      */
     public function removeFieldFromFormGroup($key)
     {
+        trigger_error('removeFieldFromFormGroup() is deprecated since version 2.4.', E_USER_DEPRECATED);
```
Comment (Soullivaneuh) [inline Admin/Admin.php]: Please update with the following message:

```php
__METHOD__.' is deprecated since 2.4 and will be removed in 3.0.'
```

File: Admin/AdminInterface.php [hunk 3]
```diff
@@ -820,6 +820,8 @@ public function setShowTabs(array $showTabs);
      * Remove a form group field.
      *
      * @param string $key
+     *
+     * @deprecated
```
Comment (Soullivaneuh) [inline Admin/AdminInterface.php]: `@deprecated since 2.4, will be removed in 3.0`

**Comment by Soullivaneuh** [comment]
Please also fix CS by running `make cs` command.

Related job: https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/68839610#L376

File: Tests/Admin/AdminTest.php [hunk 4]
```diff
@@ -1456,21 +1465,22 @@ public function testRemoveFieldFromFormGroup()
                 ),
             ),
         );
-
-        $admin = new PostAdmin('sonata.post.admin.post', 'Application\Sonata\NewsBundle\Entity\Post', 'SonataNewsBundle:PostAdmin');
         $admin->setFormGroups($formGroups);
+        $this->assertEquals($admin->getFormGroups(), $formGroups);
 
-        $admin->removeFieldFromFormGroup('foo');
```
Comment (Soullivaneuh) [inline Tests/Admin/AdminTest.php]: Deprecated method should keep being tested.

I propose:

* keep the `testRemoveFieldFromFormGroup` as is and add a `@group legacy` on doc comment.
* create a new test named `testDefineFormBuilderCleanupFormGroupsAndTabs` with your logic inside.

**Comment by EmmanuelVella** [comment]
@Soullivaneuh I have updated my PR. Thanks for your feedback !

File: Admin/Admin.php [hunk 5]
```diff
@@ -1625,6 +1658,8 @@ public function setFormGroups(array $formGroups)
      */
     public function removeFieldFromFormGroup($key)
     {
+        trigger_error(__METHOD__.' is deprecated since 2.4 and will be removed in 3.0.', E_USER_DEPRECATED);
```
Comment (Soullivaneuh) [inline Admin/Admin.php]: Sorry, I think you didn't see me comment update. :-)

```php
'The '.__METHOD__.' method is deprecated...'
```

Thanks.

**Comment by Soullivaneuh** [comment]
Not sure if `message_form_group_empty` translation key should be remove for BC.

I late @rande do the final review and merge for this one.

Thanks. :+1:

**Comment by core23** [comment]
:+1:

**Comment by Koc** [comment]
oh, no, please. Make this configurable. In our project we are having empty field group which loaded dynamically by ajax

**Comment by Soullivaneuh** [comment]
> This PR breaks our form and we should render empty fieldset by hands.

Could you elaborate?

@core23 I remove major label because it should be minor with BC.

@EmmanuelVella I'm closing this one because the wrong branch is targeted. Could you please reopen one? Thanks.
